runtime/compute/ruy/include/ruy/Shape.h

```diff
@@ -161,41 +212,38 @@ class Shape
     }
   }
 
-  // This will probably be factored out. Old code made substantial use of 4-D
-  // shapes, and so this function is used to extend smaller shapes. Note that
-  // (a) as Dims<4>-dependent code is eliminated, the reliance on this should be
-  // reduced, and (b) some kernels are stricly 4-D, but then the shapes of their
-  // inputs should already be 4-D, so this function should not be needed.
-  inline static Shape ExtendedShape(int new_shape_size, const Shape &shape)
-  {
-    return Shape(new_shape_size, shape, 1);
-  }
-
   inline void BuildFrom(const std::initializer_list<int> init_list)
   {
     BuildFrom<const std::initializer_list<int>>(init_list);
   }
 
-  // Returns the total count of elements, that is the size when flattened into a
-  // vector.
+  // Returns the total count of elements, that is the size when flattened into a vector.
   inline int FlatSize() const
   {
     int buffer_size = 1;
     const int *dims_data = DimsData();
     for (int i = 0; i < _size; i++)
     {
       const int dim = dims_data[i];
-      assert(dim >= 1);
       buffer_size *= dim;
     }
     return buffer_size;
   }
 
   bool operator!=(const Shape &comp) const { return !((*this) == comp); }
 
+  // This will probably be factored out. Old code made substantial use of 4-D
+  // shapes, and so this function is used to extend smaller shapes. Note that
+  // (a) as Dims<4>-dependent code is eliminated, the reliance on this should be
+  // reduced, and (b) some kernels are stricly 4-D, but then the shapes of their
+  // inputs should already be 4-D, so this function should not be needed.
+  inline static Shape ExtendedShape(int new_shape_size, const Shape &shape)
```

Comment:
Generally, there is no strict requirement to mark a function template as `inline`. Template functions (especially non-member ones) already avoid multiple-definition errors due to how templates are instantiated and handled by the compiler. Marking it inline usually does no harm, but it’s not strictly necessary in this situation. Nevertheless, I added it for code consistency.



In other words, you can remove the inline keyword and the function should still behave correctly.